winnowing.py

```diff
@@ -3,122 +3,123 @@
+            last = mergedPoints[-1]
+            if points[i][0] >= last[0] and points[i][0] <= last[1]: #merging overlapping regions
+                if points[i][1] > last[1]:
+                    mergedPoints = mergedPoints[:-1]
+                    mergedPoints.append([last[0], points[i][1]])
+                else:
+                    pass
+            else:
+                mergedPoints.append(points[i])
+        newCode = code[: mergedPoints[0][0]]
+        plagCount = 0
+        for i in range(len(mergedPoints)):
+            if mergedPoints[i][1] > mergedPoints[i][0]:
+                plagCount += mergedPoints[i][1] - mergedPoints[i][0]
+                newCode = newCode + '\x1b[6;30;42m' + code[mergedPoints[i][0] : mergedPoints[i][1]] + '\x1b[0m'
+                if i < len(mergedPoints) - 1:
+                    newCode = newCode + code[mergedPoints[i][1] : mergedPoints[i+1][0]]
+                else:
+                    newCode = newCode + code[mergedPoints[i][1] :]
+        return {"ratio":(plagCount/len(code)),"Code":newCode}
 
-#we form windows of hash values and use min-hash to limit the number of fingerprints
-def fingerprints(arr, winSize = 4):
-    arrLen = len(arr)
-    prevMin = 0
-    currMin = 0
-    windows = []
-    fingerprintList = []
-    for i in range(arrLen - winSize):
-        win = arr[i: i + winSize]  #forming windows
-        windows.append(win)
-        currMin = i + minIndex(win)
-        if not currMin == prevMin:  #min value of window is stored only if it is not the same as min value of prev window
-            fingerprintList.append(arr[currMin])  #reduces the number of fingerprints while maintaining guarantee
-            prevMin = currMin  #refer to density of winnowing and guarantee threshold (Stanford paper)
+    #we form windows of hash values and use min-hash to limit the number of fingerprints
+    def fingerprints(self,arr, winSize = 4):
+        arrLen = len(arr)
+        prevMin = 0
+        currMin = 0
+        windows = []
+        fingerprintList = []
+        for i in range(arrLen - winSize):
+            win = arr[i: i + winSize]  #forming windows
+            windows.append(win)
+            currMin = i + self.minIndex(win)
+            if not currMin == prevMin:  #min value of window is stored only if it is not the same as min value of prev window
+                fingerprintList.append(arr[currMin])  #reduces the number of fingerprints while maintaining guarantee
+                prevMin = currMin  #refer to density of winnowing and guarantee threshold (Stanford paper)
 
-    return fingerprintList
+        return fingerprintList
+    #takes k-gram list as input and returns a list of only hash values
+    def hashList(self,arr):
+        HL = []
+        for i in arr:
+            HL.append(i[1])
 
-#takes k-gram list as input and returns a list of only hash values
-def hashList(arr):
-    HL = []
-    for i in arr:
-        HL.append(i[1])
+        return HL
```

Comment:
This for loop can be a simple list comprehension.

Reply:
done sir